### ISSUE TITLE
Advanced Search on Hosts returns 400 if wrong variable type is passed

### ISSUE THREAD

**Issue by nixocio** [ansible/awx#9956]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY

1. Advanced Search on Hosts screen returns 400 if wrong variable type is passed with exact field select - searching for inventory - it has to be an integer.

<img width="1443" alt="image" src="https://user-images.githubusercontent.com/9053044/115412406-45b2b880-a1c2-11eb-945b-49726a6247fd.png">

On the API side:

![image](https://user-images.githubusercontent.com/9053044/115422089-90383300-a1ca-11eb-85e5-6a4bd6bdbbad.png)

2. The same does not happen to the instance group lookup.

<img width="1281" alt="image" src="https://user-images.githubusercontent.com/9053044/115418954-ed7eb500-a1c7-11eb-85e6-5c610d01c8e0.png">

On the API side.

![image](https://user-images.githubusercontent.com/9053044/115422354-d097b100-a1ca-11eb-9e57-ff9ca7026f56.png)

3. Another error case:

<img width="1251" alt="image" src="https://user-images.githubusercontent.com/9053044/115418221-531e7180-a1c7-11eb-8890-6a3edb963d55.png">

**Comment by mabashian** [comment]
I agree that the best UX would be to limit the input based on the type of value that the api is expecting (string, int, etc).  I'd actually say that case 1 and 3 are not bugs but enhancements.

Case 2 (instance group lookup) does sound like a bug to me but it looks like a bug in the api (I think).  I would expect the response to be:

```
{
    "error": ...
}
```

instead of 

```
{
    "detail": ...
}
```

This differs from the format of the error in the 1st scenario

**Comment by mabashian** [comment]
We should trim this issue down, I think.  The ask in item 1 is to provide client-side validation on fields like `id` to prevent users from performing a search that will end in an error.  That seems reasonable.  Item 2 is an API bug - we should file a separate issue.  Item 3 is covered by https://github.com/ansible/awx/issues/11469.
